src/otx/core/data/module.py

```diff
@@ -37,18 +38,26 @@ def __init__(self, task: OTXTaskType, config: DataModuleConfig) -> None:
         self.subsets: dict[str, OTXDataset] = {}
         self.save_hyperparameters()
 
-        dataset = DmDataset.import_from(
-            self.config.data_root,
-            format=self.config.data_format,
-        )
+        if self.config.data_format == "common_semantic_segmentation":
+            # temporary hack, should be fixed later
+            dataset = {}
+            train_data_roots = Path(self.config.data_root) / "train"
+            val_data_roots = Path(self.config.data_root) / "val"
+
+            for subset_name, subset_path in zip(("train", "val", "test"),
+                                           (train_data_roots, val_data_roots, val_data_roots)):
+                dataset[subset_name] = DmDataset.import_from(subset_path,
+                                                     format=self.config.data_format).subsets()["default"]
```

Comment:
I think `self.config.train_subset.subset_name` would be proper than "train". 
Please refer https://github.com/openvinotoolkit/training_extensions/pull/2685/files